tests/platform_tests/conftest.py

```diff
@@ -185,8 +185,12 @@ def service_time_check(message, status):
             timings["time_span"] = (datetime.strptime(timestamps["Started"], FMT) -\
                 datetime.strptime(timestamps["Stopped"], FMT)).total_seconds()
         elif "Start" in timestamps and "End" in timestamps:
-            timings["time_span"] = (datetime.strptime(timestamps["End"], FMT) -\
-                datetime.strptime(timestamps["Start"], FMT)).total_seconds()
+            if "last_occurence" in timings:
+                timings["time_span"] = (datetime.strptime(timings["last_occurence"], FMT) -\
+                    datetime.strptime(timestamps["Start"], FMT)).total_seconds()
```

Comment:
I think this part is covered here already?
https://github.com/Azure/sonic-mgmt/blob/master/tests/platform_tests/conftest.py#L111-L117

Reply:
I think the code you mentioned is to form the result summary, whereas this part is to make sure that the `time_span` field is calculated correctly in the first place. Also in this specific case, the part you mentioned would get skipped since `entity` is in both `time_spans` and `kexec_offsets`.

Reply:
I see, thanks for clarification. Also, why can't this `time_span` be measured as difference of End and Start times?

Reply:
`Start` and `End` time are defined as the first we see the start and end message respectively. So in the cases that the end message shows up more than once, I think we would most likely want the last occurrence. Therefore, the difference between End and Start would not cover this scenario.

Reply:
Got it. Thanks!